tests/car_tests.rs

```diff
@@ -123,10 +124,13 @@ fn new_block(rng: &mut SmallRng) -> (Cid, Vec<u8>) {
 }
 
 async fn validate_car(path: impl AsRef<Path>) -> Result<()> {
-    let mut reader = CarReader::new(tokio::fs::File::open(path).await?.compat()).await?;
-    assert!(reader.validate);
+    let mut reader = CarStream::new(tokio::io::BufReader::new(
+        tokio::fs::File::open(path).await?,
+    ))
+    .await?;
+    assert!(!reader.header.roots.is_empty());
     let mut count = 0;
```

Comment:
```suggestion
    let count = reader.count();
```

Reply:
Hmm, this does not work, `count()` takes the ownership of self. I need to call `while reader.try_next()` to validate all blocks

Reply:
So you don't actually care about the count. Looks like you just want to collect the stream with `try_collect()`. If you want the count for debugging then you can use `inspect_ok()`.